src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js

```diff
@@ -83,8 +83,10 @@ define(['config', 'sulusecurity/collections/roles'], function(Config, Roles) {
                             this.sandbox.util.each(
                                 permissionResponseData.permissions[role.id],
                                 function(index, value) {
-                                    data[index] = value;
-                                    matrixRoleData.push(value);
+                                    if (data.hasOwnProperty(index)) {
```

Comment:
This doesn't seem to be enough... The missing data will be interpreted as false.